sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/server/SparkSQLOperationManager.scala

```diff
@@ -39,15 +38,19 @@ private[thriftserver] class SparkSQLOperationManager()
   val handleToOperation = ReflectionUtils
     .getSuperField[JMap[OperationHandle, Operation]](this, "handleToOperation")
 
-  val sessionToActivePool = Map[SessionHandle, String]()
-  val sessionToContexts = Map[SessionHandle, SQLContext]()
+  val sessionToActivePool = new ConcurrentHashMap[SessionHandle, String]()
```

Comment:
It's only `private[thriftserver]`. It's minor, and a whole lot of stuff in Spark that should be `private` isn't, but I wondered if it was worth it here because you're concerned with synchronizing access to this object and therefore possibly concerned with what is accessing it. The usages you changed look like they're sufficiently protected, but are there others BTW?